### PR TITLE
rust: update dependencies

### LINKED ISSUE #13157: wasm: Should update cranlift and wastime.
We have in rust/Cargo.lock cranelift-codegen version 0.92.0 and wasmtime 5.0.0.

Dependabot (see the "security" tab) warned us that we should upgrade to 0.92.1 and 5.0.1, respectively. The versions we use today have a vulnerability which can in theory be used by a malicious UDF to read *some* of the memory outside the sandbox.

### PR DESCRIPTION
**Pull request by wmitros**
Cranelift-codegen 0.92.0 and wasmtime 5.0.0 have security issues potentially allowing malicious UDFs to read some memory outside the wasm sandbox. This patch updates them to versions 0.92.1 and 5.0.1 respectively, where the issues are fixed.

Fixes #13157

### CODE REVIEW

**Comment by wmitros** [comment]
The changes ended up being the same as in https://github.com/scylladb/scylladb/pull/13166, but the commit message should be clearer here

**Review by nyh — APPROVED** [review]
Looks good. I'll wait for the CI before merging.

**Comment by nyh** [comment]
By the way, this vulnerability makes me think that even after UDFs and wasm stops being experimental, we should add separate option enabling and disabling the different UDF languages - instead of all of them together with `--enable-user-defined-functions` (which IIUC today enables all the languages). For example, if a Scylla installation uses UDF but only in Lua, it shouldn't need to worry about vulnerabilities suddenly being discovered in Wasm and some low-privilege malicious application may get privilege escalation by sending some malicious Wasm code.

**Comment by nyh** [comment]
> The changes ended up being the same as in #13166, but the commit message should be clearer here

I guess that all these packages have interdependencies anyway, so the fixed wasmtime depends on the fixed cranelift-codegen, so just updating wasmtime was enough to get everything updated.

**Pull request closed without merging**
